bigml/lda.py

```diff
@@ -0,0 +1,310 @@
+# -*- coding: utf-8 -*-
+#!/usr/bin/env python
+#
+# Copyright 2016 BigML
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+"""A local Predictive LDA Model.
+
+This module allows you to download and use LDA models for local
+predicitons.  Specifically, the function LDA.distribution allows you
+to pass in input text and infers a generative distribution over the
+topics in the learned LDA model.
+
+Example usage (assuming that you have previously set up the BIGML_USERNAME
+and BIGML_API_KEY environment variables and that you own the lda/id
+below):
+
+from bigml.api import BigML
+from bigml.lda import LDA
+
+api = BigML()
+
+lda = LDA('lda/5026965515526876630001b2')
+topic_distribution = lda.distribution({"text": "A sample string"}))
+
+"""
+from bigml.api import FINISHED
+from bigml.api import (BigML, get_lda_id, get_status)
+from bigml.util import cast, utf8
+from bigml.basemodel import retrieve_resource
+from bigml.basemodel import ONLY_MODEL
+from bigml.model import STORAGE
+from bigml.modelfields import ModelFields
+
+import random
+import logging
+import array
+import Stemmer
+
+LOGGER = logging.getLogger('BigML')
+
+MAX_TERM_LENGTH = 30
+UPDATES = 64
+
+CODE_TO_NAME = {
+    "da": u'danish',
+    "nl": u'dutch',
+    "en": u'english',
+    "fi": u'finnish',
+    "fr": u'french',
+    "de": u'german',
+    "hu": u'hungarian',
+    "it": u'italian',
+    "nn": u'norwegian',
+    "pt": u'portuguese',
+    "ro": u'romanian',
+    "ru": u'russian',
+    "es": u'spanish',
+    "sv": u'swedish',
+    "tr": u'turkish'
+}
+
+class LDA(ModelFields):
+    """ A lightweight wrapper around a cluster model.
+
+    Uses a BigML remote cluster model to build a local version that can be used
+    to generate centroid predictions locally.
```

Comment:
ditto: changing cluster to LDA and centroid to distribution or topic distribution